eureka_ml_insights/data_utils/aime_utils.py

```diff
@@ -24,26 +20,16 @@ def parse_output_answer(response):
     """
     Parse the input string to extract answer of a given AIME question.
     Parameters:
-    s (str): Input string containing answer X in the form of "Final Answer: X".
-    Returns:
-    dict: A numeric value representing the model's output.
-          Example: 255
+        response (str): Input string containing answer X in the form of "Final Answer: X".
+    Returns: 
+        numerical_value (float): A numeric value representing the model's answer.
     """
+    numerical_value = None
+
     # Try to find an answer in the "Final Answer: X" format
     match = re.search(r"Final Answer:\s*([\$]?-?[\d,]+(?:\.\d+)?%?)", response)
     if match:
         answer_str = match.group(1)
-    else:
-        # If that fails, look for any number with optional $ or % in the response
-        match = re.search(r"([\$]?-?[\d,]+(?:\.\d+)?%?)", response)
-        answer_str = match.group(1) if match else None
-    # return float(answer_str)
-    numerical_value = None
-    # Store the original format
-    original_format = answer_str
-
-    print("----answer string", answer_str)
-    if answer_str:
```

Comment:
May I ask for the reason to remove this backup parsing?

Reply:
discussed offline